src/ssh.rs

```diff
@@ -70,38 +69,52 @@ impl Handler {
     pub async fn new() -> io::Result<Handler> {
         let permit = CONCURRENCY.acquire().await.unwrap();
         let mask = find_available_token();
-        let name = format!("{}-{}\0", FILE_MAP_NAME, mask);
+        let name = HSTRING::from(format!("{}-{}\0", FILE_MAP_NAME, mask));
         let handle = unsafe {
-            CreateFileMappingA(
+            CreateFileMappingW(
                 INVALID_HANDLE_VALUE,
-                ptr::null_mut(),
+                Some(ptr::null()),
                 PAGE_READWRITE,
                 0,
                 PUTTY_IPC_MAXLEN as u32,
-                name.as_str(),
+                &name,
             )
         };
-        if handle.is_null() {
-            release_token(mask);
-            return Err(other_error(format!(
-                "failed to create memory mapping: {}",
-                Error::last_os_error()
-            )));
-        }
+
+        let handle = match handle {
+            Ok(handle) => handle,
+            Err(_e) => {
+                release_token(mask);
+                return Err(other_error(format!(
+                    "failed to create file mapping: {}",
+                    Error::last_os_error()
+                )));
+            }
+        };
+
         let view = unsafe { MapViewOfFile(handle, FILE_MAP_ALL_ACCESS, 0, 0, PUTTY_IPC_MAXLEN) };
-        if view.is_null() {
-            unsafe {
-                CloseHandle(handle);
+
+        let view = match view {
+            Ok(view) => view,
+            Err(_e) => {
+                unsafe {
+                    CloseHandle(handle);
+                }
+                release_token(mask);
+                return Err(other_error(format!(
+                    "failed to map view of file: {}",
+                    Error::last_os_error()
```

Comment:
Why use last os error instead of the error returned by `MapViewOfFile`?

Reply:
Right, that's a better way to put it.